src/constant_evaluator/mod.rs

```diff
@@ -91,6 +91,15 @@ impl<'a> Evaluator<'a> {
                 let values = &self.other_constants[name.as_str()];
                 values[abstract_to_degree(&arg_values[0]) as usize % values.len()].clone()
             }
+            Expression::MatchExpression(scrutinee, arms) => {
+                let v = self.evaluate(scrutinee);
+                for (n, e) in arms {
+                    if n.is_none() || Some(&v) == n.as_ref() {
+                        return self.evaluate(e);
+                    }
+                }
+                panic!("No arm matched the value {v}");
```

Comment:
```suggestion
                arms.iter()
                    .find(|(n, _)| !matches!(n, Some(n) if *n != v))
                    .map(|(_, e)| self.evaluate(e))
                    .expect("No arm matched the value {v}")
```

Reply:
I find the negated matches macro quite hard to read to be honest.

Reply:
Maybe `find (|(n, _)| n == None || n.as_ref() == Some(&v))`?